From c2c3872d867d62be1a43a6d2c3007cc1c26af2c6 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Sat, 14 May 2016 17:38:24 -0400
Subject: [PATCH 0438/1239] thunderx: Drop stale FDT retrieval code

Since this hacky call is replaced with a cleaner method of
FDT passing, drop it

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 board/cavium/thunderx/fdt.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/board/cavium/thunderx/fdt.c b/board/cavium/thunderx/fdt.c
index 2f85fd193e..d6effc49f7 100644
--- a/board/cavium/thunderx/fdt.c
+++ b/board/cavium/thunderx/fdt.c
@@ -18,6 +18,8 @@
 
 #define MAX_LMAC_PER_BGX 4
 
+DECLARE_GLOBAL_DATA_PTR;
+
 static const void *get_prop_value(void *fdt, const char *prop_name, int *len)
 {
 	int depth = 0, node;
@@ -69,17 +71,15 @@ void thunderx_parse_bdk_config(void)
 {
 	char boardname[32];
 	const char *str;
-	void *fdt = (void *)CONFIG_BDK_FDT_START;
 	int ret = 0, len = 32;
 
-	atf_get_bdk_fdt(fdt, CONFIG_BDK_FDT_SIZE);
-	if (fdt != NULL) {
-		ret = fdt_check_header(fdt);
+	if (gd->fdt_blob != NULL) {
+		ret = fdt_check_header(gd->fdt_blob);
 		if (ret < 0) {
 			printf("fdt: %s\n", fdt_strerror(ret));
 		} else {
-			debug("fdt:size %d\n", fdt_totalsize(fdt));
-			str = get_prop_value(fdt, "BOARD-MODEL", &len);
+			debug("fdt:size %d\n", fdt_totalsize(gd->fdt_blob));
+			str = get_prop_value(gd->fdt_blob, "BOARD-MODEL", &len);
 			debug("fdt: str %s len %d\n", str, len);
 			if (str) {
 				strncpy(boardname, str, len);
@@ -87,7 +87,7 @@ void thunderx_parse_bdk_config(void)
 			} else {
 				printf("Err: cannot retrieve board type from fdt\n");
 			}
-			thunderx_parse_phy_address(fdt);
+			thunderx_parse_phy_address(gd->fdt_blob);
 		}
 	}
 
-- 
2.29.0

